pkg/resources/rdma/mlnxRdma.go

```diff
@@ -0,0 +1,38 @@
+package rdma
+
+import (
+	"github.com/Mellanox/rdmamap"
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1"
+
+	"github.com/intel/sriov-network-device-plugin/pkg/types"
+)
+
+// mlnxRdmaDir path to rdma resources
+const mlnxRdmaDir = "/dev/infiniband"
+
+type mlnxRdmaSpec struct {
+	isRdma     bool
+	deviceSpec []*pluginapi.DeviceSpec
+}
+
+// NewMlnxRdmaSpec returns the Mellanox vendor RdmaSpec
+func NewMlnxRdmaSpec(pciAddrs string) types.RdmaSpec {
+	isRdma := len(rdmamap.GetRdmaDevicesForPcidev(pciAddrs)) > 0
+	deviceSpec := make([]*pluginapi.DeviceSpec, 0)
+	if isRdma {
+		deviceSpec = append(deviceSpec, &pluginapi.DeviceSpec{
+			HostPath:      mlnxRdmaDir,
+			ContainerPath: mlnxRdmaDir,
+			Permissions:   "rwm",
+		})
+	}
+	return &mlnxRdmaSpec{isRdma: isRdma, deviceSpec: deviceSpec}
+}
+
+func (r *mlnxRdmaSpec) IsRdma() bool {
+	return r.isRdma
+}
+
+func (r *mlnxRdmaSpec) GetRdmaDeviceSpec() []*pluginapi.DeviceSpec {
+	return r.deviceSpec
+}
```

Comment:
@Mmduh-483 there is nothing mellanox vendor specific done here, rename it to rdma/rdma.go

Reply:
rdma/<vendor>Rdma.go is for vendor-specific RDMA if the vendor wants to initialize, configure, ...etc. anything for RDMA it will be here, @moshe010 @ahalim-intel is it good to add another function here like 
`initRdmaDeviceSpec` or `setupRdmaDeviceSpec`

Reply:
Any init/setup can be done within the `New` itself. Is there a need that init/setup be called from outside this file?